### PR TITLE
chore: bump rc-dialog & rc-image version

### LINKED ISSUE #40810: [Modal] 对话框希望增加一个切换动画结束后的回调
### What problem does this feature solve?

因为对话框在动画执行过程中渲染页面获取的宽度不够准确导致会时不时的布局错乱。所以希望跟 Drawer抽屉一样增加一个 afterOpenChange 切换动画结束后的回调

### What does the proposed API look like?

afterOpenChange: (open: boolean) => void

### PR DESCRIPTION
**Pull request by MuxinFeng**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [x] Other (about what?)

### 💡 Background and solution
close https://github.com/ant-design/ant-design/issues/40810

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      Upgrade rc-dialog, rc-image versions, support modal components to pass afterOpenChange props     |
| 🇨🇳 Chinese |   升级 rc-dialog、rc-image 版本，支持 Modal 组件传入 afterOpenChange props        |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by MuxinFeng** [comment]
rebase 操作出了问题，原 pr 被关了 https://github.com/ant-design/ant-design/pull/40852#

**Comment by li-jia-nan** [comment]
要先更新 rc-image 里面的 rc-dialog 版本吧

**Comment by MuxinFeng** [comment]
> 要先更新 rc-image 里面的 rc-dialog 版本吧

这样啊，我去看看

**Comment by li-jia-nan** [comment]
- https://github.com/react-component/image/pull/200

**Comment by li-jia-nan** [comment]
那边刚刚升上去了，你再推一下吧，让 CI 重跑一下

**Comment by MuxinFeng** [comment]
> 要先更新 rc-image 里面的 rc-dialog 版本吧

是在失败的 ci details 里看的吧，antd 这边是要求底层依赖都保持同步吗

**Comment by yoyo837** [comment]
同时升级一下  "rc-image": "~5.16.0",

File: components/modal/index.en-US.md [hunk 1]
```diff
@@ -70,6 +70,7 @@ Additionally, if you need show a simple confirmation dialog, you can use [`App.u
 | zIndex | The `z-index` of the Modal | number | 1000 |  |
 | onCancel | Specify a function that will be called when a user clicks mask, close button on top right or Cancel button | function(e) | - |  |
 | onOk | Specify a function that will be called when a user clicks the OK button | function(e) | - |  |
+| afterOpenChange | Callback when the animation ends when Modal is turned on and off | function(open) | - |  |
```
Comment (li-jia-nan) [inline components/modal/index.en-US.md:73]: 这个 open 是布尔值吗？可以把类型也加上

File: components/modal/index.zh-CN.md [hunk 2]
```diff
@@ -71,6 +71,7 @@ demo:
 | zIndex | 设置 Modal 的 `z-index` | number | 1000 |  |
 | onCancel | 点击遮罩层或右上角叉或取消按钮的回调 | function(e) | - |  |
 | onOk | 点击确定回调 | function(e) | - |  |
+| afterOpenChange | 打开和关闭 Modal 时动画结束后的回调 | function(open) | - |  |
```
Comment (li-jia-nan) [inline components/modal/index.zh-CN.md:74]:
```suggestion
| afterOpenChange | 打开和关闭 Modal 时动画结束后的回调 |(open: boolean) => void | - | 5.3.2 |
```

Comment (MuxinFeng) [reply components/modal/index.zh-CN.md:74]: 好嘞，我统一改吧，这样分支管理舒服一点

**Comment by li-jia-nan** [comment]
PR title 也改一下，还有 rc-image

**Review by li-jia-nan — APPROVED** [review]

**Comment by afc163** [comment]
TS 定义不对：https://github.com/ant-design/ant-design/issues/41735

**Pull request merged**